src/streaming/utils/TTMLParser.js

```diff
@@ -71,6 +73,133 @@ function TTMLParser() {
     }
 
     /**
+     * Get the begin-end interval if present, or null otherwise.
+     *
+     * @param {Object} element - TTML element which may have begin and end attributes
+     */
+    function getInterval(element) {
+        if (element.hasOwnProperty('begin') && element.hasOwnProperty('end')) {
```

Comment:
@nigelmegitt I realized what you meant before you answered. However, the code actually supports both the sidecar and embedded TTML cases. In the sidecar case, there is no sample timing, so that is the reason I have the test and only do cropping when the intervalStart and intervalEnd are available.